testsuite/tests/typing-local/alloc.ml

```diff
@@ -456,6 +456,15 @@ let[@inline never] huge () =
   bytes_set b pos 'h';
   assert (bytes_get b pos = 'h')
 
+class cla = object
+    val x = 42
+end
+
+let[@inline never] obj () =
+  ignore_local (new cla);
+  ()
```

Comment:
Could you also add a direct creation of an object with the `object ... end` syntax?